src/native/minipal/utf8.h

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_UTF8_H
+#define HAVE_MINIPAL_UTF8_H
+
+#include <minipal/utils.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+#define MINIPAL_MB_ERR_INVALID_CHARS 0x00000008
```

Comment:
How does the error reporting work with this flag? 

It would be nice to include the error reporting in the doc comment below.

Reply:
Now that most of the errors are changed to asserts, we don't error out. Renamed to `MINIPAL_MB_NO_REPLACE_INVALID_CHARS`.

Reply:
What is the behavior when the input contains invalid chars and this flag is specified?

Reply:
Before addressing 'turn errors into asserts': https://github.com/dotnet/runtime/pull/85558#discussion_r1211076625, it was setting errno and failing, now it asserts in Debug and produce incorrect result in Release.

Reply:
> produce incorrect result in Release.

This is not good behavior. Is this flag actually needed? Are there any tests failing when this flag is not respected?

We should either delete this flag; or keep returning error for invalid chars when this flag is set.

Reply:
I had that error condition earlier which I changed based on your feedback: https://github.com/dotnet/runtime/pull/85558#discussion_r1211076625. Retesting this stuff isn't very straight forward. Lets postpone cosmetic stuff for the later PR if necessary.

Reply:
I am not sure that this is cosmetic issue. The assert is reachable via invalid input. For example, `./ildasm -METADATA invalid.dll` where invalid.dll is hand-edited to contain invalid UTF8 sequence fails with this change with assert:
```
ildasm: /home/jkotas/runtime/src/native/minipal/utf8.c:344: _Bool FallbackInvalidByteSequence_Copy(UTF8Encoding *, unsigned char **, CHAR16_T **): Assertion `self->useFallback' failed.
```


My earlier feedback was about the ContractAsserts that cannot fail due to invalid user input. 

This error condition is different. It can fail due to invalid user input and it was not actually handled by ContractAsserts in the original code.

Reply:
> I am not sure that this is cosmetic issue.

Currently, all tests are passing with both main and the PR and we do have invalid characters cases in Reflection.Emit, PAL and other tests.

> For example, `./ildasm -METADATA invalid.dll` where invalid.dll is hand-edited

Can you provide that invalid test case? I will like to compare main vs. PR behavior.

Reply:
Here is repro case: [repro.zip](https://github.com/dotnet/runtime/files/11793894/repro.zip)

It is standard hello world, with first 4 bytes of `Program` replace with `F0 00 80 80` in hex editor.

Reply:
Thanks, now it has error handling for invalid sequence.

PR:
```
Assertion failed: (self->useFallback), function FallbackInvalidByteSequence_Copy, file utf8.c, line 344.
zsh: abort      artifacts/bin/coreclr/osx.arm64.Debug/ildasm -METADATA ~/Downloads/repro.dll
```
main / PR+fix
```
WARNING: Out of memory condition being issued from: /Users/am11/projects/runtime/src/coreclr/md/compiler/regmeta_import.cpp, line 754

GetTypeDefProps failed.
Failed return code: 0x8007000e
```